modules/scram/scram.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+class ScramAuthenticator {
+
+    private $hashes = array(
+        'sha-1'   => 'sha1',
+        'sha1'    => 'sha1',
+        'sha-224' => 'sha224',
+        'sha224'  => 'sha224',
+        'sha-256' => 'sha256',
+        'sha256'  => 'sha256',
+        'sha-384' => 'sha384',
+        'sha384'  => 'sha384',
+        'sha-512' => 'sha512',
+        'sha512'  => 'sha512'
+    );
+
+    private function getHashAlgorithm($scramAlgorithm) {
+        $parts = explode('-', strtolower($scramAlgorithm));
+        return $this->hashes[$parts[1]] ?? 'sha1'; // Default to sha1 if the algorithm is not found
+    }
+    private function log($message) {
+        echo "[DEBUG] " . $message . "\n";
```

Comment:
Please use Hm_Debug class instead of echo to standard output - look how it is done in other places.